Classes/Notifications/NewFeaturesController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  NewFeaturesController.swift
+//  Freetime
+//
+//  Created by Ryan Nystrom on 11/22/18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+import GitHubAPI
+import IGListKit
+
+final class NewFeaturesController {
+
+    var version: String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    private var fetchURL: URL? {
+        return URLBuilder(host: "raw.githubusercontent.com")
+            .add(paths: ["GitHawkApp", "Release-Notes", "master", "versions", "\(version).md"])
+            .url
+    }
+
+    private let session: URLSession = {
+        let config = URLSessionConfiguration.default
+        config.requestCachePolicy = .reloadIgnoringLocalCacheData
+        config.urlCache = nil
+        return URLSession.init(configuration: config)
+    }()
+
+    private let userDefaultsKey = "com.freetime.new-features-controller.has-displayed"
+
+    var hasDisplayedLatest: Bool {
+        get {
+            guard let last = UserDefaults.standard.string(forKey: userDefaultsKey) else {
+                return false
+            }
+            // value kept to latest version when seen
+            // pinned to old versions (or empty) when out of date
+            return last >= version
+        }
+        set {
+            if newValue {
+                UserDefaults.standard.set(version, forKey: userDefaultsKey)
+            } else {
+                UserDefaults.standard.removeObject(forKey: userDefaultsKey)
+            }
+        }
+    }
+
+    func fetch(completion: @escaping (Result<String>) -> Void) {
+        guard let url = fetchURL else { return }
```

Comment:
Same with this guy, maybe add at least an `assert`?

Reply:
Recently at work I checked this by doing a search and replace for `else { return }` and replacing it with an `else { return print("\(#file):\(#line) returned early") }`.

Should write a quick blog post about that :)

Reply:
And you can do the same for `else { return nil }` with `else { print(""); return nil }`